File: Backend/Application/DTO/RuleDto.cs
using Domain.Enums;

namespace Application.DTO
{
    public class RuleDto
    {
        public Guid Id { get; set; }
        public string Keyword { get; set; } = null!;
        public MatchTypes MatchType { get; set; }
        public ActionType Action {  get; set; }
        public string? Color { get; set; }
        public string? Label { get; set; }
        public bool IsEnable { get; set; }
        public RulePriority Priority { get; set; }
    }
}

File: frontend/src/Interfaces/Rule/RuleDto.ts
import type { ActionType } from "./ActionType";
import type { MatchTypes } from "./MatchTypes";
import type { RulePriority } from "./RulePriority";

export interface RuleDto{
    id: string;
    keyword: string;
    matchType: MatchTypes;
    action:ActionType;
    color:string | null;
    label:string | null;
    isEnable: boolean;
    priority:RulePriority;
}